src/utilities/ad.jl

```diff
@@ -1,14 +0,0 @@
-using Zygote: pullback
-using ProximalCore
-
-struct ZygoteFunction{F}
-    f::F
-end
-
-(f::ZygoteFunction)(x) = f.f(x)
-
-function ProximalCore.gradient!(grad, f::ZygoteFunction, x)
```

Comment:
The diff would be much smaller if you kept your functions `gradient` and `gradient!` but made them rely on `value_and_pullback` instead, right?

Reply:
I guess, but I’m thinking that these primitives don’t really fit the interface of ProximalCore maybe, see https://github.com/JuliaFirstOrder/ProximalCore.jl/pull/3